src/Config/HotReloadEventHandler.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config
+{
+    /// <summary>
+    /// HotReloadEventHandler manages the events that are needed to signal refreshing
+    /// classes that must be updated during a hot reload.
```

Comment:
Need a comment here describing why events are defined in this class instead of the RuntimeConfigLoader class. 

From .NET Event Programming Guide: https://learn.microsoft.com/en-us/dotnet/standard/events/#events
> The event is typically a member of the event sender; 

In this case, we can't do that because of this other behavioral issue that would arise:https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-raise-base-class-events-in-derived-classes

> When you create a class that can be used as a base class for other classes, you should consider the fact that events are a special type of delegate that can only be invoked from within the class that declared them.

-> Adding this event handler class as the class to be injected via DI to all dependent services instead of each of the dependent services loaded "RuntimeConfigLoader" class.

Reply:
Good observation! @aaronburtle it might be helpful to add this in the code comments/PR description..

Reply:
added comment to try and clarify this.